### ISSUE TITLE
Add SCOORD to a Measurement in SR (TID 320)

### ISSUE THREAD

**Issue by Fedalto** [ImagingDataCommons/highdicom#306]
Hi, is it possible to add spatial coordinates to a Measurement in SR?

I'm looking at [TID 320 Image or Spatial Coordinates](https://dicom.nema.org/medical/dicom/current/output/chtml/part16/chapter_A.html#sect_TID_320), specifically row 3 and 4.

If it's not currently possible, I could work on a PR to add it.

But I could use some pointers on how to best implement it.
As I understand, the `referenced_images` parameter in `sr.Measurement` implements the first row of TID 320 (inferred from image).
What's the best solution to add the other options? Should a new parameter in `sr.Measurement` be added for each one? Or the `referenced_images` type could be changed to `Optional[Sequence[SourceImageForMeasurement | SourceScoordForMeasurement]]`?

**Comment by CPBridge** [comment]
Hi @Fedalto, thanks for the question

This is not supported right now but I would certainly consider a PR to add it.

However, have you considered whether using already supported behaviour of [PlanarROIMeasurementsAndQualitativeEvaluations](https://highdicom.readthedocs.io/en/latest/package.html#highdicom.sr.PlanarROIMeasurementsAndQualitativeEvaluations) (or [VolumetricROIMeasurementsAndQualitativeEvaluations](https://highdicom.readthedocs.io/en/latest/package.html#highdicom.sr.VolumetricROIMeasurementsAndQualitativeEvaluations) if you want SCOORD3D coordinates) could be used for your use case? In these cases you can specify the `referenced_region` (or `referenced_regions`) as `highdicom.sr.ImageRegion` when constructing the template, along with `measurements` that go along with it.

So this is sort of the the "other way up" to what you propose. You specify an image region and then, within it, measurements that relate that region. Would that work for your use case? I would like to better understand in what scenarios TID 320 would be prefereable.

**Comment by Fedalto** [comment]
Hi @CPBridge.

I'm using the ImageRegion as well. \
My understanding of SR is that I can create a "TID 1411 Volumetric ROI Measurements and Qualitative Evaluations" that represents an ROI, and inside it I use many ImageRegions (1 for each axial slice) to define a 3D contour of the ROI. I then add many measurements for that ROI using the "TID 300. Measurement" template. One of those measurements could be a Diameter. What I want to do is define the diameter line, the coordinates of the start and end points. That looks possible using the row 3 of TID 320 ("INFERRED FROM SCOORD") and that will also link to the image with "SELECTED FROM". Here, I'm not sure when "R-SELECTED FROM" could be used instead.

So, in the end, the SR structure would be like 
```
└── ROI (TID 1411)
    ├── 1..n Image Regions that defines the contour of the ROI
    └── 1..n Measurements like a diameter (TID 300)
        └── INFERRED FROM SCOORD like MULTIPOINT x1,y1,x2,y2
            └── SELECTED FROM IMAGE, here a reference to the original image
```

**Comment by CPBridge** [comment]
Resolved by #307
